### PR TITLE
New -kv-version argument to force version and avoid mounts call

### PR DESCRIPTION
**Pull request by adrianlzt**
Added a new argument "-kv-version" that could be used with the "kv" subcommands to force the KV engine version to be used.

Forcing the version allows the CLI to skip the call to /v1/sys/internal/ui/mounts/kv.
This allow the CLI to get cached vaules from a proxy without intervention of the server.

Related to #19879

### CODE REVIEW

**Comment by adrianlzt** [comment]
`vault read my-kv-v2-mount/data/foo` does not call `/v1/sys/internal/ui/mounts/kv`

**Comment by adrianlzt** [comment]
I think something similar could be useful to be able to access the cache of an offline proxy.

The autocomplete needs also some changes, as it also calls ``sys/mounts``.

Maybe could be possible to make ``sys/mounts`` cacheable?

File: command/kv_helpers.go [hunk 1]
```diff
@@ -119,7 +119,11 @@ allow read access to that path in the format 'mount-path/data/foo', not just 'mo
 	return mountPath, 1, nil
 }
 
-func isKVv2(path string, client *api.Client) (string, bool, error) {
+func isKVv2(path string, client *api.Client, kvVersion int) (string, bool, error) {
+	if kvVersion != 0 {
+		return strings.SplitN(path, "/", 2)[0], kvVersion == 2, nil
```
Comment (VioletHynes) [inline command/kv_helpers.go:124]: What's the reason behind `SplitN` here? Wouldn't `strings.Split(path, "/")[0]` be equivalent?

File: command/kv_get.go [hunk 2]
```diff
@@ -168,6 +168,7 @@ func (c *KVGetCommand) Run(args []string) int {
 		}
 	}
 
+	// Print current date with ms precision
```
Comment (VioletHynes) [inline command/kv_get.go:171]: Why was this comment added? Can this be removed?

File: command/base.go [hunk 3]
```diff
@@ -550,6 +553,19 @@ func (c *BaseCommand) flagSet(bit FlagSetBit) *FlagSets {
...
 
+		// FlagSetKVCommon
+		if bit&FlagSetKVCommon != 0 {
+			kvCommonSet := set.NewFlagSet("KV Common Options")
+
+			kvCommonSet.IntVar(&IntVar{
+				Name:    "kv-version",
+				Target:  &c.flagKVVersion,
+				Default: 0,
+				Usage:   `Specifies the version of the KV backend to use. If not set it will be detected.`,
```
Comment (VioletHynes) [inline command/base.go:564]:
```suggestion
				Usage:   `Specifies the version of the KV backend to use. If not set, it will be detected with an API call.`,
```

File: command/kv_delete.go [hunk 4]
```diff
@@ -76,10 +76,10 @@ func (c *KVDeleteCommand) Flags() *FlagSets {
 		Name:    "mount",
 		Target:  &c.flagMount,
 		Default: "", // no default, because the handling of the next arg is determined by whether this flag has a value
-		Usage: `Specifies the path where the KV backend is mounted. If specified,
```
Comment (VioletHynes) [inline command/kv_delete.go:79]: Please remove these stylistic changes to strings throughout the PR.

**Review by VioletHynes — COMMENTED** [review]
Looks good so far. A few suggestions, and if you're still interested in getting this merged, I'd like to see some tests using the new option, exercising both kvv1 and kvv2 paths, and some edge cases.

**Comment by VioletHynes** [comment]
I'm going to close this, as it seems like the user got a solution for their issue and this PR is no longer needed, as feedback has lingered for a while. If interested in merging this PR, please feel free to open another PR :)
